includes/wikia/parser/templatetypes/TemplateTypesParser.class.php

```diff
@@ -20,7 +20,9 @@ public static function onFetchTemplateAndTitle( &$text, &$finalTitle ) {
 			switch ( $type ) {
 				case AutomaticTemplateTypes::TEMPLATE_NAVBOX:
 				case TemplateClassificationService::TEMPLATE_NAVBOX:
-					$text = self::handleNavboxTemplate();
+				case TemplateClassificationService::TEMPLATE_NAV:
+				case TemplateClassificationService::TEMPLATE_FLAG:
+					$text = self::handleHiddenTemplate();
```

Comment:
Yeah, I originally did that and went back and forward a few times thinking on what is more useful: having types with shared behaviour/treatment or having handling per template purpose. I finally came down on treatment, but I was very conflicted. ;) I'll go back to separate per purpose as it does provide a little more flexibility going forward.